std/traits.d

```diff
@@ -5628,12 +5635,28 @@ enum bool isInstanceOf(alias S, T) = is(T == S!Args, Args...);
     static struct Bar(T...) { }
     static struct Doo(T) { }
     static struct ABC(int x) { }
+    static void fun(T)() { }
+    template templ(T) { }
+
     static assert(isInstanceOf!(Foo, Foo!int));
     static assert(!isInstanceOf!(Foo, Bar!int));
     static assert(!isInstanceOf!(Foo, int));
     static assert(isInstanceOf!(Doo, Doo!int));
     static assert(isInstanceOf!(ABC, ABC!1));
-    static assert(!__traits(compiles, isInstanceOf!(Foo, Foo)));
+    static assert(!isInstanceOf!(Foo, Foo));
```

Comment:
I guess it's because the original implementation did not handle things different from types as second parameter, and `Foo` is not a type, so it wouldn't compile.